devops/docker/CIDockerfile

```diff
@@ -7,8 +7,12 @@ ENV DOCKER_SHA256_x86_64 340e0b5a009ba70e1b644136b94d13824db0aeb52e09071410f35a9
 
 RUN apt-get update && \
     apt-get install -y flake8 make virtualenv ccontrol libpython2.7-dev \
-            libffi-dev libssl-dev libyaml-dev python-pip curl git &&\
-    apt-get clean
+            libffi-dev libssl-dev libyaml-dev python-pip curl git npm &&\
+    apt-get clean && \
+    # We are safe to do `npm install --global` because html_codesniffer has no
+    # dependencies, this line makes sure that stays true.
+    npm view html_codesniffer 2>/dev/null | head -n 2 | cut -d'|' -f 3 | grep none && \
```

Comment:
Nice!  It looks like you can simplify this even further by [asking for just the `dependencies` field](https://docs.npmjs.com/cli/v7/commands/npm-view#description) and checking its length.  Compare:

```sh-session
$ npm view html_codesniffer dependencies
$ npm view express dependencies | head
 
{
  accepts: '~1.3.8',
  'array-flatten': '1.1.1',
  'body-parser': '1.20.1',
  'content-disposition': '0.5.4',
  'content-type': '~1.0.4',
  cookie: '0.5.0',
  'cookie-signature': '1.0.6',
  debug: '2.6.9',
```

Reply:
@saffronsnail, I've realized in conversation with @legoktm that even this `npm view` check is overkill.  _Since_ `html_codesniffer@2.5.1` has no dependencies, it's fine just to pin `npm install` to that known-good version.  Since a ["specific name and version combination can never be reused"][unpublish], installing `html_codesniffer@2.5.1` will _never_ pull any dependencies, even if a subsequent version should add some.  (If that happens, we're back in `package-lock.json` territory as previously discussed.)

Pinning to a version we know to be good, which here means "has no dependencies", is the more-general solution to this problem, rather then explicitly checking for no dependencies.

[unpublish]: https://docs.npmjs.com/cli/v9/commands/npm-unpublish

Reply:
Fixed!